### PR TITLE
*: update ks3 SDK and add some workaround for ks3 SDK

### LINKED ISSUE #48278: Disable SSL when connect to self-managed S3 buckets
## Enhancement
according to https://www.vldb.org/pvldb/vol16/p2769-durner.pdf , we can turn off SSL to reduce CPU cost

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #48278 close https://github.com/pingcap/tidb/issues/48318

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/importer/import.go [hunk 1]
```diff
@@ -931,14 +931,34 @@ func (e *LoadDataController) InitDataStore(ctx context.Context) error {
...
+		s, err = e.initSelfManagedExternalStore(ctx, cloudStorageURL, target)
 		if err != nil {
 			return err
 		}
 		e.GlobalSortStore = s
 	}
 	return nil
 }
+
+func (*LoadDataController) initSelfManagedExternalStore(ctx context.Context, u *url.URL, target string) (storage.ExternalStorage, error) {
```
Comment (D3Hunter) [inline pkg/executor/importer/import.go:943]: what you mean by `self-managed`?

Comment (lance6716) [reply pkg/executor/importer/import.go:943]: s3 does not belong to user, it belongs to our cloud infra

**Review by D3Hunter — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**